quick_sort/QuickSort.java

```diff
@@ -1,9 +1,42 @@
 public class QuickSort {
 
+    /*
+    This method returns the index of median of array at i, j, k.
+    @param arr: Input array to be sorted.
+    @param i: Left end index of array.
+    @param j: Right end index of array.
+    @param k: Middle index of array.
+    @return: Index of median of array at i, j, k.
+    */
+    private static int median(int[] arr, int i, int j, int k) {
+        if (arr[i] > arr[j] && arr[i] > arr[k]) {
+            if (arr[k] > arr[j]) {
+                return k;
+            } else {
+                return j;
+            }
```

Comment:
Make a function for this : 
```java
if (arr[k] > arr[j]) {
                return k;
            } else {
                return j;
            }
```

Reply:
@aashutoshrathi Making a separate function for such small work wouldn't be good as it will increase the time of execution by extra function calls.

Reply:
But writing the same code for 3 times? Is it good? 😕